### ISSUE TITLE
object detection : matchin outputs.last_hidden_state with results

### ISSUE THREAD

**Issue by fenaux** [huggingface/transformers#39426]
### Feature request

it seems to me that would be possible with a little modification in the function post_process_object_detection

with
```
``for score, label, box, index in zip(scores, labels, boxes, indexes):
            results.append(
                {
                    "scores": score[score > threshold],
                    "labels": label[score > threshold],
                    "boxes": box[score > threshold],
                    "indexes": index[score > threshold],
                }
            )``
```
 and then 
`outputs.last_hidden_state[0][results[0]['indexes']] `
gives me the desired vector features

Am I right or is there a better way to obtain this matching ? 

Thanks for your help

### Motivation

I would like to use outputs.last_hidden_state as features for auxiliary tasks. So I need to know the label and the bounding box associated to one given vector of outputs.last_hidden_state

### Your contribution

I am not a top coder and do not know how to submit a PR

**Comment by qubvel** [comment]
Hey @fenaux, `outputs.logits` and `outputs.pred_boxes` are used to obtain the results in `post_process_object_detection`

**Comment by fenaux** [comment]
@qubvel, sorry but your comment doesn't explain to match the results to the embeddings. Let's say if i want to select embeddings of one given class as obtained after post_process_object_detection

**Comment by qubvel** [comment]
Hi @fenaux, sorry, maybe I did not get what you mean, you can explore the modeling file to get a better understanding, e.g. `src/transformers/detr/modeling_detr.py`

https://github.com/huggingface/transformers/blob/main/src/transformers/models/detr/modeling_detr.py

**Comment by fenaux** [comment]
Hi @qubvel 
In order to apply a tracker (for instance https://github.com/mikel-brostrom/boxmot/tree/master) the common format is to have detection 
- where detection is (bounding box, confidence, class). This is given by results.
- and the appearance of this detection being given by an embedding vector (very often the last hidden state of Clip or SigLip or a detection model)

Of course the appearance vector as to be associated with the right detection. So my question / proposal aims to find a way to make this matching without having to recompute every thing that has already been done in post_process_object_detection

I hope this help clarify

**Comment by qubvel** [comment]
In that case, I would recommend writing a custom post-processing step based on `post_process_object_detection` to preserve the order of the features and detections

I suppose, the simplest case would be for DETR model

https://github.com/huggingface/transformers/blob/a44dcbe513e3e073271e0b8e369b75aca51affae/src/transformers/models/detr/image_processing_detr.py#L1775-L1826
